tests/JunitFormatterTest.php

```diff
@@ -177,9 +178,13 @@ private function prettyPrintXml(string $inputXml): string
         return trim($outputXml);
     }
 
+    /**
+     * @throws DOMException
```

Comment:
Exceptions from other namespaces should never bubble up. Catch it at the lowest possible level and wrap to some our exception. That one can be propagated.

Doing that, you benefit from PHPStan exception analysis, see `checkedExceptionClasses`